### ISSUE TITLE
Resurrect `findowner` to help update OWNERS files

### ISSUE THREAD

**Issue by dims** [kubernetes/community#3700]
Originally when we created OWNERS files, we used https://github.com/coreos/findowner to seed the reviewers and approvers etc. 

Can we please pick that up again and run it periodically?

**Comment by dims** [comment]
cc @nikhita

**Comment by nikhita** [comment]
> Can we please pick that up again and run it periodically?

I'm still looking at the tool...out of curiosity, what did you have in mind about how we should use it? From a very first glance, it looks to me as a tool that approvers should be running for the "path" they are interested in updating reviewers for (the tool has a default depth of 3 from the given path).

**Comment by dims** [comment]
@nikhita essentially i'd like contribex to drive the updates of at least reviewers (add new folks, delete inactive folks) on a periodic basis. We could update the tool however we see fit, i don't have any specific preferences there. The problem i am trying to address is that it's hard for folks to propose themselves via PRs so they hesitate which is not good for the health of the project. Same problem with inactive reviewers getting piled up with newer PRs which end up getting stale as well. (i am sure you know all this!). I was looking at what was used when we bootstrapped the OWNERS effort and found this tool.

**Comment by nikhita** [comment]
> The problem i am trying to address is that it's hard for folks to propose themselves via PRs so they hesitate which is not good for the health of the project.

This is so true, thanks for talking about this!

This looks like a great idea, and I'm going to try playing around with tool at least with the community repo.

> essentially i'd like contribex to drive the updates of at least reviewers (add new folks, delete inactive folks) on a periodic basis. 

Do you mean updating reviewers even for k/k?

cc @parispittman @cblecker @Phillels

**Comment by dims** [comment]
yes, even for k/k.

**Comment by spiffxp** [comment]
I've been toying around with a few scripts in this direction myself, based on PR reviews vs. code commits.  Nothing scalable enough to hit the whole codebase though

https://gist.github.com/spiffxp/97ee57e348d27c59e029ce55d11a50a6

**Comment by nikhita** [comment]
> yes, even for k/k.

I think sig contribex shouldn't be owning updating OWNERS files in k/k. It _is_ related to "contributor experience" per se, but I feel like each SIG should be responsible for updating their own OWNERS files. We (sig contribex) can tell them about best practices and tools but not own updating other SIGs' owners files.

**Comment by dims** [comment]
@nikhita we could propose new reviewers. it would be up to them to merge the PR. would that work?

**Comment by dims** [comment]
Clearly very few SIG(s) are doing what's necessary at the moment ... Hence asking.

**Comment by spiffxp** [comment]
ref: https://github.com/kubernetes/community/issues/1808

**Comment by mml** [comment]
Want to mention again that I think there are two dimensions to this problem:

1. Are the existing members of the OWNERS file good choices?  That is, do they have the right context, judgement, and trust of the community?  I had a concrete proposal for this in #2878 which I think is workable.
2. Let's make sure reviews get appropriate levels of attention and that our throughput and latency don't suffer.  I discussed this a bit in https://github.com/kubernetes/community/issues/2873#issuecomment-434126158 and as I pointed out, if we focus on PRs instead of reviewers, we can decide, for example, to focus our efforts to maximize the positive impact on contributors and possibly even on new contributors.

This seems like a useful tool for growing a list or for making sure we include newer contributors in OWNERS files, especially if they wouldn't otherwise self-nominate.  If that's the main issue here (removing a barrier for newer contributors), then maybe we could re-title this issue in terms of that rather than a specific tool?
